### PR TITLE
Fix emojione 3 issues

### PR DESCRIPTION
**Pull request by johnsenpeder**
When using emojione 3 some emojis are failing to render after save.

It seems to fail on https://github.com/johnsenpeder/emojionearea/blob/b8b684195957e5843adc0d45ee2d2c00c98ae81b/src/function/getTemplate.js#L29 when using `unicode.uc_base` (results in `✈` => ✈). However it works fine when using `unicode.uc_output.toUpperCase()` (results in `✈️` => ✈️).

Also cleaned up some trailing whitespaces and converted a few tabs to spaces. (I could create separate PR's if preferred)

### CODE REVIEW

File: dist/emojionearea.js [hunk 1]
```diff
@@ -76,10 +76,17 @@ document = window.document || {};
         if (shortname) {
             friendlyName = shortname.substr(1, shortname.length - 2).replace(/_/g, ' ').replace(/\w\S*/g, function(txt) { return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();});
         }
+        var fname = '';
+        if (emojioneSupportMode > 4) {
```
Comment (lasdeparq) [inline dist/emojionearea.js]: ` if (unicode.uc_base && emojioneSupportMode > 4) {`

Comment (lasdeparq) [reply dist/emojionearea.js]: In some cases the method getTemplate receive unicode var as String, and throws an error when try to access uc_base.

Fixing this point, works correctly

File: package.json [hunk 2]
```diff
@@ -28,9 +28,9 @@
   },
   "homepage": "https://github.com/mervick/emojionearea#readme",
   "dependencies": {
-    "emojione": ">=2.0.1",
-    "jquery": ">=1.8.3",
-    "jquery-textcomplete": ">=1.3.4"
+    "emojione": "^3.1.2",
+    "jquery": "^3.2.1",
+    "jquery-textcomplete": "^1.8.4"
```
Comment (lasdeparq) [inline package.json]: Do not change the dependencies, because emojionearea can work with old versions too.

File: package.json [hunk 3]
```diff
@@ -28,9 +28,9 @@
   },
   "homepage": "https://github.com/mervick/emojionearea#readme",
   "dependencies": {
-    "emojione": ">=2.0.1",
-    "jquery": ">=1.8.3",
-    "jquery-textcomplete": ">=1.3.4"
+    "emojione": "^3.1.2",
+    "jquery": "^3.2.1",
```
Comment (lasdeparq) [inline package.json]: Do not change the dependencies, because emojionearea can work with old versions too.

File: package.json [hunk 4]
```diff
@@ -28,9 +28,9 @@
   },
   "homepage": "https://github.com/mervick/emojionearea#readme",
   "dependencies": {
-    "emojione": ">=2.0.1",
-    "jquery": ">=1.8.3",
-    "jquery-textcomplete": ">=1.3.4"
+    "emojione": "^3.1.2",
```
Comment (lasdeparq) [inline package.json]: Do not change the dependencies, because emojionearea can work with old versions too.

File: package.json (hunk 2 above)
Comment (johnsenpeder) [reply package.json]: @lasdeparq didn't even realise these were changed :), will revert :+1:

File: dist/emojionearea.js (hunk 1 above)
Comment (johnsenpeder) [reply dist/emojionearea.js]: @lasdeparq will update :+1:

**Comment by johnsenpeder** [comment]
@lasdeparq all updated :)

**Comment by mervick** [comment]
Thanks)

**Pull request merged**